src/test/regress/expected/isolation_metadata_sync_vs_all.out

```diff
@@ -0,0 +1,595 @@
+Parsed test spec with 3 sessions
+
+starting permutation: s1-begin s2-begin s1-start-metadata-sync s2-alter-table s1-rollback s2-rollback s3-compare-snapshot
+create_distributed_function
+---------------------------------------------------------------------
+
+(1 row)
+
+step s1-begin:
+    BEGIN;
+
+step s2-begin:
+ BEGIN;
+
+step s1-start-metadata-sync:
+ SELECT start_metadata_sync_to_node('localhost', 57638);
+
+start_metadata_sync_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s2-alter-table:
+ ALTER TABLE dist_table ADD COLUMN z int;
+ <waiting ...>
+step s1-rollback: 
+    ROLLBACK;
+
+step s2-alter-table: <... completed>
+step s2-rollback:
+ ROLLBACK;
+
+step s3-compare-snapshot:
+ SELECT bool_and(result::text[] @> activate_node_snapshot() AND result::text[] <@ activate_node_snapshot()) AS same_snapshot_across_cluster
+ FROM run_command_on_workers('SELECT activate_node_snapshot()');
+
+same_snapshot_across_cluster
+---------------------------------------------------------------------
+t
+(1 row)
+
+master_remove_node
+---------------------------------------------------------------------
+
+
+(2 rows)
+
+
+starting permutation: s1-begin s2-begin s1-start-metadata-sync s2-drop-table s1-rollback s2-rollback s3-compare-snapshot
+create_distributed_function
+---------------------------------------------------------------------
+
+(1 row)
+
+step s1-begin:
+    BEGIN;
+
+step s2-begin:
+ BEGIN;
+
+step s1-start-metadata-sync:
+ SELECT start_metadata_sync_to_node('localhost', 57638);
+
+start_metadata_sync_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s2-drop-table:
+ DROP TABLE dist_table;
+ <waiting ...>
+step s1-rollback: 
+    ROLLBACK;
+
+step s2-drop-table: <... completed>
+step s2-rollback:
+ ROLLBACK;
+
+step s3-compare-snapshot:
+ SELECT bool_and(result::text[] @> activate_node_snapshot() AND result::text[] <@ activate_node_snapshot()) AS same_snapshot_across_cluster
+ FROM run_command_on_workers('SELECT activate_node_snapshot()');
+
+same_snapshot_across_cluster
+---------------------------------------------------------------------
+t
+(1 row)
+
+master_remove_node
+---------------------------------------------------------------------
+
+
+(2 rows)
+
+
+starting permutation: s1-begin s2-begin s1-start-metadata-sync s2-create-dist-table s1-rollback s2-rollback s3-compare-snapshot
+create_distributed_function
+---------------------------------------------------------------------
+
+(1 row)
+
+step s1-begin:
+    BEGIN;
+
+step s2-begin:
+ BEGIN;
+
+step s1-start-metadata-sync:
+ SELECT start_metadata_sync_to_node('localhost', 57638);
+
+start_metadata_sync_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s2-create-dist-table:
+ CREATE TABLE new_dist_table(id int, data int);
+ SELECT create_distributed_table('new_dist_table', 'id');
+ <waiting ...>
+step s1-rollback: 
+    ROLLBACK;
```

Comment:
Why do we ROLLBACK in the tests? Wouldn't it be better to `COMMIT` and `s3-compare-snapshot`? Such that we ensure the current operation is done successfully? '

Valid for all rollback steps in the test -- unless there are any specific ROLLBACKs

Reply:
I commit everything now.